---
#title: "Differential analisys"
#author: "Alexey Uvarovskii"
#date: "October 18, 2017"
output: 
  revealjs::revealjs_presentation: 
    css: ../styles.css
    center: true
 #   incremental: true
    transition: fade
#    mathjax: local
    self_contained: false
    lib_dir: ../libs
#    reveal_plugins: ["chalkboard"]
    reveal_options:
      chalkboard:
        theme: whiteboard
  html_document: default
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Differential expression

Alexey Uvarovskii

17-10-17
<div>
<img src="fig/heart.png" 
  style = "height: 100px; border:none; box-shadow: none;" >
</div>

-----------
### DE analysis

>- RNA level (other feature)
>- Groups

<div class="fragment" style="padding-top:100px">
which genes/tx/exons are affected?
</div>

```{r include = FALSE}
knitr::opts_chunk$set(cache = TRUE, echo = FALSE, message = FALSE,
                      warning = FALSE)
```

----------

### What is a difference


```{r echo = FALSE, fig.width=3, fig.height=2, dpi = 400, out.width="600px"}
cols <- list(
rgb(255, 255, 227,  maxColorValue = 255),
rgb(49, 67, 71,  maxColorValue = 255),
rgb(108, 150, 98, maxColorValue = 255),
rgb(222, 107, 72, maxColorValue = 255),
rgb(145, 47, 86, maxColorValue = 255))

library(cowplot)
ggplot(data = data.frame(x = c("Control", "K.O."), y = c(10, 50))) +
  geom_col(aes(x = x, y = y), width = .4)+ 
  xlab("") + ylab("CXCR5")
```

## Uncertainty
Compare your confidence if you

-  made 20 qPCR replicates 
-  made 1 replicate (*the other did not work*)
-  got the numbers from a guy in Irish pub

------------------

<h3> Variance defines conclusion</h3>

```{r}
knitr::opts_chunk$set(fig.height= 3, fig.width = 4,
                      out.width = "600px", dpi =400)
```

```{r fig.width = 8, fig.height=3, out.width="900px", dpi = 400}

set.seed(239)
x1 <- rnbinom(100,mu = 10, size = 1)
x2 <- rnbinom(100, mu = 50, size = 2)

d <- data.frame(x = c(x1, x2), group = c(rep("Control",100), rep("KO", 100)))
q1 <- qplot(
  data = d,
  x = x,
  size = I(1.3),
  colour = group,
  geom = "density",
  xlab = "CXCR5"
  ) + 
  scale_color_manual(values = c("dodgerblue", "tomato"))

x1 <- rnbinom(100,mu = 10, size = 100)
x2 <- rnbinom(100, mu = 50, size = 200)
d2 <- data.frame(x = c(x1, x2), group = c(rep("Control",100), rep("KO", 100)))
q2 <- qplot(
  data = d2,
  x = x,
  size = I(1.3),
  colour = group,
  geom = "density",
  xlab = "CXCR5"
  ) + 
  scale_color_manual(values = c("dodgerblue", "tomato"))
cowplot::plot_grid(q1,q2)
```

--------------

### Variance and counts

```{r echo = TRUE}
replicatesNumber <- 100
lowExpression    <- 10
highExpression   <- 10000
# simulate experiment ('random' + 'Poisson' = rpois)
rnaLow           <- rpois(replicatesNumber, lowExpression)
rnaHigh          <- rpois(replicatesNumber, highExpression)
# what is proportion of st. dev.?
sd(rnaLow)  / mean(rnaLow)
sd(rnaHigh) / mean(rnaHigh)
```

-------------

### Low count measurements are noisy

```{r fig.width=4, fig.height=4, out.width="500px", dpi = 400, echo=TRUE}
boxplot(rnaLow / lowExpression, rnaHigh / highExpression,
        names = c("low", "high"))
     
```

------------
### Linear models

[expression control] = [base level]   <br> 
[expression KO] = [base level] + [effect]

<div class = "fragment" style = "color:blue">
Is effect = 0?
</div>

------------

### Multifactorial design

[RNA] = [base] + [KO eff.] + [Treatm]

<div class = "fragment">
 control without treatment <br>
[RNA] = [base] + <span style = "color:red">0</span>[KO eff.] +
<span style = "color:red">0</span>[Treatm]
</div>
<div class = "fragment">
 KO with treatment <br>
[RNA] = [base] + 
<span style = "color:red">1</span>
[KO eff.] + 
<span style = "color:red">1</span>[Treatm]
</div>


-----------

### No Interaction

```{r fig.width=4, fig.height=3, out.width="700px", dpi = 400}

d <- data.frame(
  cxcr5 = c(50, 100, 150, 200),
  group = c("Cont", "KO", "Cont.Treat", "KO.Treat"))
ggplot() + 
  geom_point(data = d,aes(x = group, y = cxcr5)) +
  scale_y_continuous(limits = c(0,250)) + 
  geom_line(aes(x = c(2,2), y = c(50,150))) + 
  geom_line(aes(x = c(4,4), y = c(100,200))) + 
  geom_line(aes(x = 1:2, y = 50), linetype = 2) + 
  geom_line(aes(x = 3:4, y = 100), linetype = 2) + 
  ylab  ("CXCR5") + 
  ggtitle("Same treatmenr effect") +
  xlab("")
```


-------------

### Interaction

```{r fig.width=4, fig.height=3, out.width="700px", dpi = 400}

d <- data.frame(
  cxcr5 = c(50, 100, 150, 120),
  group = c("Cont", "KO", "Cont.Treat", "KO.Treat"))
ggplot() + 
  geom_point(data = d,aes(x = group, y = cxcr5)) +
  scale_y_continuous(limits = c(0,250)) + 
  geom_line(aes(x = c(2,2), y = c(50,150))) + 
  geom_line(aes(x = c(4,4), y = c(100,120))) + 
  geom_line(aes(x = 1:2, y = 50), linetype = 2) + 
  geom_line(aes(x = 3:4, y = 100), linetype = 2) + 
  ylab  ("CXCR5") + 
  ggtitle("KO responds differently") +
  xlab("")
```

-----------

### Interaction in model 

 KO responds differently to treatment <br>
[RNA] = [base] + 
<span style = "color:red">1</span>
[KO eff.] + 
<span style = "color:red">1</span>[Treatm] +
<span style = "color:green">1[Treatm.KO]</span>

-------

### Normalisation

<div align="left">
CXCR5 in <br>
Run 1: 1000 reads <br>
Run 2: 2000 reads (2x higher depth)
<div class = "fragment">
All genes have double expression?
</div>
</div>

--------------

### FPKM / RPKM / CPM 
<br>
<div align="left">
$$\text{ CPM} =  \frac{\text{reads}}{\text{depth }} $$
$$\text{ RPKM} =  \frac{\text{reads}}{\text{depth }\cdot \text{length }} $$
</div>
<br>
*depth* in mio's, *length* in kB <br>
FPKM is RPKM, only F = fragments

-------

### When it fails

```{r fig.width=6, fig.height=3, out.width="800px", dpi = 400}
d <- data.frame(
  CXCR5 = c(50,100),
  CXCR4 = c(30, 30),
  group = c("Cont", "KO")
)
qplot(
  data = reshape2::melt(d),
  x = variable,
  y = value,
  geom = "col",
  facets = ~ group,
  xlab = "",
  ylab = "", 
  fill = variable
  ) +
  scale_fill_manual(name = "",values = c("dodgerblue", "tomato")) + 
  theme( strip.background = element_rect(fill = "skyblue"),
         axis.line.y  = element_line(color = "skyblue"),
         axis.line.x  = element_line(color = "skyblue"))
```

<div class = "fragment">
 It is not the library size, it is  overexpressed genes
 </div>

-------

### Robust normalisation

example

```{r echo = TRUE}
sample1 <- c(
  cxcr5 = 10,
  cxcr4 = 20,
  ccr7  = 30,
  ccl2  = 40,
  pd1   = 50
)
sample2 <- 2 * sample1 # sequencing depth x 2
# normalise
rbind(
sample1 = sample1 / sum(sample1),
sample2 = sample2 / sum(sample2))
```

----------------

### Add high changes

```{r echo = TRUE}
sample2[5] <- 300 # high expression
rbind(
sample1 = sample1 / sum(sample1),
sample2 = sample2 / sum(sample2))

```

---------

### Use robust normalisation

```{r echo = TRUE}
rbind(
sample1 = sample1 / sum(sample1[ sample1 < quantile(sample1,.5) ]),
sample2 = sample2 / sum(sample2[ sample2 < quantile(sample2,.5) ]))
```

-----------

### Why it works

only rank is important, not value:

```{r echo = TRUE}
median(c(1,2,3,4,5))
median(c(1,2,3,4,500))
```

---------

### Batch effects

---------

### PCA

--------

### Clustering

----

### Annotation

------

### Variations

------

DExon usage

------

## Resources
[R installation](https://github.com/Bioconductor/BiocIntro/blob/R-Intro-RPCI-Jan-2017/vignettes/A1_Installation.Rmd)  
[workflow 1](https://www.bioconductor.org/help/course-materials/2017/OSU/B3_RNASeq_Workflow.html)  
[workflow 2](https://bioconductor.org/help/workflows/rnaseqGene/)
