<!DOCTYPE html>

<html prefix="og: http://ogp.me/ns#">
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
		
    <meta  property='og:title' content="pulseR for RNA kinetics"/>
    <meta  property='og:image' content="alexey0308.github.io/tk/pulseR/fig/social.jpg"/>
    <meta  property='og:image' content="alexey0308.github.io/tk/pulseR/fig/social-2.jpg"/>
    
    <meta  property='og:description' content="An R package for analysis of RNA-seq in pulse-chase experiments."/>
    <meta  property='og:url' content="//alexey0308.github.io/tk/pulseR/slides.html" />
    <meta name="twitter:image" content="alexey0308.github.io/tk/pulseR/fig/social.jpg" />


		<title>pulseR for RNA kinetics</title>

		<link rel="stylesheet" href="reveal.js/css/reveal.css">
		
		 <!-- <link rel="stylesheet" href="hd.css" id="theme"> -->
     <link rel="stylesheet" href="pulseR.css" id="theme">


		<!-- Theme used for syntax highlighting of code -->
	      <link rel="stylesheet" href="reveal.js/lib/css/github.css"> 
    <link rel="stylesheet" href="reveal.js/lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
      var link = document.createElement( 'link' );
      link.rel = 'stylesheet';
      link.type = 'text/css';
      link.href = window.location.search.match( /print-pdf/gi ) ? 'reveal.js/css/print/pdf.css' : 'reveal.js/css/print/paper.css';
      document.getElementsByTagName( 'head' )[0].appendChild( link );
    </script>
      <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-103088620-1', 'auto');
    ga('send', 'pageview');

  </script>

	</head>
	<body>
	
		<div class="reveal">
<div style="position:absolute;  bottom:1%; left:1%; height: 60px; "> 
<img src="fig/siegel/logo-final.png" height="100%" style="vertical-align:middle;">
<span style="font-size:.6em; z-index:1;"> Alexey Uvarovskii.  pulseR for RNA kinetics </span>
</div>
		
			<div class="slides">
			
<section id="main" class="slide level1" data-background-color="#333333"
data-background-transition="slide" data-transition="slide" data-state="title">
<p style="font-family: sans" >Analysis of RNA kinetics <br> 
<span style="font-size: 70%" </span> using</p>
<!-- <div style = "left: 250px; top: 200px; position: absolute" > -->
<div>
  <img  src="logo_background.gif" width="300">
</div>
<!-- <div style = "left: 250px; top: 400px; position: absolute" > -->
<div>
  <div style="display: inline-block">
    <p style="line-height: 1; font-family: sans; font-size: 70%" >
    Alexey Uvarovskii,  Christoph Dieterich</p>
    <p style="font-family: sans; font-size: 70%">
      University Hospital Heidelberg </p>
  </div>
<!--
  <div style="display: inline-block;">
    <img align="center" style="background-color: #333333" src="heart.png" width="100">
  </div>
-->
</div>
</section>

<section data-transition="slide">
<div>
  <h3 > Acknowledgement </h3>
</div>
<div style="">
  <div style="display: inline-block; vertical-align: middle">
    <img align="center" src="fig/siegel/siegel-uni.jpeg" width="200" >
  </div>
  <div style="display: inline-block; vertical-align: middle">
    <img align="center" src="fig/siegel/bmbf.png" width="250">
  </div>
  <div style="display: inline-block; vertical-align: middle">
    <img align="center" src="fig/siegel/Klaus-Tschira-Stiftung_Logo.jpg" width="400">
  </div>
</div>
<p style="font-size:.8em">
 Tobias Jakobi (Dieterich Lab) - computing support </p>
<p style="font-size:.8em">
 David Vilchez, Seda
Koyuncu (CECAD Cologne), <br> Janine Altmüller, Marek Franitza (CCG Cologne) - 
experimental data
 </p>
</section>

<section data-transition="slide-in ">
<h2> Is there a difference? </h2>
<div style="position:relative; width:640px; margin:0 auto;">
<p class="fragment " data-fragment-index="0" style="font-size:80%; ">
Number of motor vehicle deaths in the US <span style="font-size:80%"> (wiki) 
</span></p>
</div>
<div style="position:relative; width:640px; height:480px; margin:0 auto;">
  <img class="fragment fade-out" data-fragment-index="0"  height="480" width="640px"
  src="fig/twoBar.png" style="position:absolute;top:0; left:0;" />
  <img class="fragment "  data-fragment-index="0"  height="480" width="640px"
  src="fig/twoBar-cars.png" style="position:absolute;top:0; left:0;" /> 
</div>
</section>


<section>
<h2> Same number, different context </h2>
<div class="fragment" >
<h2> I bet you think about</h2> 
<h1 style="color:#4690d4"> Rates </h1>
</div>
</section>

<section> 
<h2> Reads in RNA-seq </h2>
<div style="position:relative; width:640px; margin:0 auto;">
<p  style="font-size:80%; visibility: hidden;">
Number of motor vehicle deaths in the US <span style="font-size:80%"> (wiki) 
</span></p>
</div>
<div style="position:relative; width:640px; height:480px; margin:0 auto;">
  <img   height="480" width="640px" src="fig/twoBar-rna.png" style="position:absolute;top:0;left:0;" />
</div>
</section>

<!-- red color:#e31a1c -->
<!-- blue color:#4690d4 -->
<section > 
<h2> RNA level is a balance </h2>
<p >
<span class="fragment " style="font-size:1.2em" data-fragment-index="1" >
  $\sf \frac{d[\text{RNA}]}{dt} = +$</span>
<span style="color:#e31a1c" >$\sf  [\text{synthesis}] $ </span>
<span class="fragment" data-fragment-index="1" >$\sf  -$ </span>
<span  style="color:#4690d4" >$\sf  [\text{degradation}]$</span>
<span class="fragment" data-fragment-index="1" >$\sf \cdot [\text{RNA}]$</span>
</p>
<p style="font-family:sans" class="fragment" data-fragment-index="1" >
$$ \sf [\text{steady state RNA}] = \frac{[\text{synthesis}]}{[\text{degradation }]}$$
</p>
</section>

<section >
<h3> Nascent RNA can be traced </h3>
<p style="font-size".8em"> EU = 5-ethyniluridine </p>
<div style="position:relative; width:540px;   margin:0 auto;">
  <img src="fig/fisher2.svg" width="540px" > <br>
  <span style="font-size:.3em"> from the Click-iT® Kit
  Thermofisher manual </span>
</div>
usually followed by  RNA-seq
</section>

<section >
<h3> Pulse-chase experiment </h3>
<p> a way to measure RNA kinetics </p>
<div>
    <img align="center" src="fig/chase.png" width="520">
</div>
</section>

<section> 
<h2> Background </h2>
<ul>
<li> rates are also interesting </li>
<li> new RNA can be traced </li>
<li> pulse-chase RNA-seq </li>
</ul>

</section>

<section>
<h1> Analysis </h1>
<ul style="margin: 0 0 30px 0">
<li> kinetic model </li>
<li> stat model </li>
<li> normalisation </li>
</ul>

<div class="fragment" >
<h3> 
<span style="font-family: monospace"> pulseR </span> to help </h3>
<p style="font-size:.5em">
Alexey Uvarovskii, Christoph Dieterich; pulseR: Versatile computational analysis of RNA turnover from metabolic labeling experiments. Bioinformatics 2017 btx368. doi: 10.1093/bioinformatics/btx368
</p>
</div>
</section>

<section>
<h2> Kinetic model </h2>
<p style="padding: 0 0 40px 0"> defined by the setup,
 e.g. pulse labelling is </p>
<p> $$
\begin{align} \sf [\text{total}] &= T\equiv\text{const} \\
 \sf[\text{pull down}] &= T\cdot \left( 1 - e^{-dt}\right) 
\end{align}$$
</p>
<p>
$$\sf t =  1, 2, 4\, \text{hr} $$
</p>
</section>

<section>
<h2> Stat model </h2>
<p> Negative binomial distribution </p>
<img src="fig/overdispersion.jpg" height="300px">
<p style="font-size:.6em">
purple: no overdispersion, yellow: with overdispersion
</p>
<p style="font-size:.4em">
Anders, Simon, and Wolfgang Huber. 
"Differential expression analysis for sequence count data."
Genome biology 11.10 (2010): R106.
</p>
</section>

<section>
<h3> Normalisation </h3>
$$\begin{align}\sf
[\text{total}]  &= \sf\,[\text{labelled}] + [\text{unlabelled}] \\
\sf[\text{pull down}] &=\sf ?[\text{labelled}] + ?[\text{unlabelled}] 
\end{align}
$$
</section>

<section>
<h3> Normalisation using spike-ins </h3>
<div>
    <img align="center" src="fig/spikeins.png" width="650">
</div>
</section>

<section >
<h3> In <span style="font-family: monospace"> pulseR </span></h3> 
<h4> there are two options for normalisation: </h4>
<div style="display: inline-block; text-align: right;">
<p>  using spike-ins (DESeq)</p>
<p> by MLE fitting </p> 
</div>
<div style="display: inline-block; text-align: left">
 <p class="fragment" data-fragment-index="1" style="color:#e31a1c">absolute synthesis rate</p>
 <p class="fragment" data-fragment-index="1" style="color:#e31a1c">no spike-ins needed</p>
</div>
</section>

<section >
<h3> Alternatives </h3> 
<div style="position:relative; height:360px;  margin:0 auto;">
<img src="fig/table-color.png" height="360px"     style="vertical-align: middle;">
</div>
<p style="font-size:.7em">
 N: normal, NB: negative binomial, BIN: binomial. 
</p>
</section>



<section >
<h3> EU pulse-chase on H9 cells </h3>

<div style="position:relative; width:640px; height: 480px; margin:0 auto;">
  <div style="position:absolute; top:0; left:0; "> 
    <img src="fig/fit.png" width="640px">
  </div>
  <div style="position:absolute;  top:00px; right:-250px; z-index:1;  "> 
   <p style="font-size:.5em; margin: 0 0 0 0;position: relative; left: -20px;top: 20px;" >design</p>
    <img src="fig/vilchez.png" width="300px">
  </div>
</div>

</section>

<section >
<h3> A number is not enough </h3>
<p>
<span style="font-family: monospace"> pulseR </span> 
can estimate confidence intervals for you 
</p>
<div style="display: inline-block; text-align: left">
 <p class="fragment" data-fragment-index="1" style="color:#e31a1c">
 diagnostics and comparisons</p>
</div>


</section>

<section >
<h3> Confidence intervals </h3>
<div style="position:relative; width:640px; height:480px; margin:0 auto;">
  <img class="fragment fade-out" data-fragment-index="0" width="640" 
  src="fig/ciH9.png" style="position:absolute;top:0;left:0;" />
  <img class="fragment "  data-fragment-index="0" width="640" 
  src="fig/ciH9-rect.png" style="position:absolute;top:0;left:0;" />
</div>
</section>

<section>

<h3> Profile likelihood </h3>
$ \frac{\log L(d)}{\log L(d_{optimal})} < \frac{1}{2}\chi^2_{0.95,1 \,d.f.} \approx 1.92 $
<div style="position:relative;  height:400px; margin:0 auto;">
<img src="fig/goodCI.png" height="100%">
</div>
</section>

<section>
<h3> Uncertainty in fit </h3>
<div style="position:relative;  height:400px; margin:0 auto;">
<img src="fig/ci.gif" height="100%">
</div>

</section>
        

<section class="slide level1">
  <h3> The workflow </h3>
 <pre class="r">
 <code >
library(pulseR)
# put math here
formulas <- MeanFormulas(
  total = mu,
  labelled = mu * (1 - exp(-d*22)) * exp(-d*time),
  unlabelled = mu * (1 - exp(-d*time) * (1 - exp(-d * 22)))
)
# define the fractions
formulaIndexes <- list(
  total_fraction = 'total',
  pull_down      = c('labelled', 'unlabelled'))
 </code>
 </pre>
 <pre class="r">
 <code data-trim>
pd <- PulseData(counts, conditions, formulas, formulaIndexes,
  groups = ~ fraction + time)
result <- fitModel(pd, initValues, opts)
 </code>
 </pre>
  </section>
  
<section id="outline" class="slide level1">
<h3>
<span style="font-family: monospace"> pulseR </span> allows to 
</h3>
<ul>
<li>estimate kinetic rates from RNA-seq</li>
<li> flexible analysis (spike-ins, cross-contamination, etc.)</li>
<li> diagnostics with profile likelihood</li>
</ul>
</section>


<section data-background-color="#ffffff">

<p style="top:0px;  "> Poster A-271 </p>
<p> An open post-doc position </p>
<div>

<div style=" text-align: left;">
  <div style="width:80%">
    <div style=" display: inline;">
      <img src="heart.png" width="100px" style="vertical-align: middle">
    </div>
    <div style=" display: inline;">
      <span>dieterichlab.org </span> 
    </div>
  </div>
  <div style="width:80%">
    <div style=" display: inline;">
      <img src="logo.png" width="100px" style="vertical-align: middle">
    </div>
    <div style="width:20%; display: inline;">
      <span>github.com/dieterich-lab/pulseR </span>
    </div>
  </div>
</div>

</div>

<div> 
<p style="font-family:monospace"> a.uvarovskii@uni-heidelberg.de </p>
</div>
</section>


    </div>
  </div>

		<script src="reveal.js/lib/js/head.min.js"></script>
		<script src="reveal.js/js/reveal.js"></script>

		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
        transition: 'none',
        backgroundTransition: 'fade',
        //width: 1600,
    //height: 900,
          math: {
         //mathjax: 'https://cdn.mathjax.org/mathjax/latest/MathJax.js',
         //mathjax: 'reveal.js/lib/js/MathJax.js',
        config: 'TeX-AMS_HTML-full'  // See http://docs.mathjax.org/en/latest/config-files.html
        },
                 slideNumber: 'c',
				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
					{ src: 'reveal.js/plugin/markdown/marked.js' },
					{ src: 'reveal.js/plugin/markdown/markdown.js' },
					{ src: 'reveal.js/plugin/notes/notes.js', async: true },
					{ src: 'reveal.js/plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
        { src: 'reveal.js/plugin/math/math.js', async: true }
				]
			});
            Reveal.addEventListener( 'fragmentshown', function( event ) {
                var video = event.fragment.querySelector( 'video' );
                if( video) {
                    video.play();
                }
            } );
      //Reveal.addEventListener( 'slidechanged', function( event ) {
      //  MathJax.Hub.Rerender();
      //});
		</script>
		

	
	
	<!-- footers -->
	<style type="text/css">
    /* 1. Style header/footer <div> so they are positioned as desired. */
    #header-left {
        position: absolute;
        top: 0%;
        left: 0%;
    }
    #header-right {
        position: absolute;
        top: 0%;
        right: 0%;
    }
    #footer-left {
        position: absolute;
        bottom: 0%;
        left: 0%;
    }
</style>


	</body>
</html>
